pkg/kubectl/cmd/set/set_port.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+// PortOptions is the start of the data required to perform the operation. As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type PortOptions struct {
+	resource.FilenameOptions
+
+	Mapper            meta.RESTMapper
+	Typer             runtime.ObjectTyper
+	Infos             []*resource.Info
+	Encoder           runtime.Encoder
+	Selector          string
+	Out               io.Writer
+	Err               io.Writer
+	DryRun            bool
+	ShortOutput       bool
+	All               bool
+	Record            bool
+	Output            string
+	ChangeCause       string
+	Local             bool
+	Cmd               *cobra.Command
+	port              int
+	protocol          string
+	overwrite         bool
+	containerSelector string
+	portsForObject    func(object runtime.Object) ([]string, error)
+
+	PrintObject            func(cmd *cobra.Command, isLocal bool, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error
+	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
+}
+
+var (
+	portResources = `
```

Comment:
Should we add like this? This does not seems like k8s style.

Please refer below,

https://github.com/kubernetes/kubernetes/blob/ea017719e52138263cb6e82eff8652cdd465d322/pkg/kubectl/cmd/cmd.go#L202-L242

But we're trying to replace such hard-coded way, refer to #42873 and #51324.

@liggitt PTAL. For updating such container ports, can we use the old hard-coded strings.

Reply:
this mean support update env's resources